src/cwt/claim/ietf_token_status.rs

```diff
@@ -0,0 +1,151 @@
+//! Claims from the IETF Token Status List spec.
+
+use super::{Claim, Error as ClaimError, Key};
+use serde::{Deserialize, Serialize};
+use serde_cbor::Value;
+use std::collections::BTreeMap;
+
+/// Status Claim for the ReferencedToken.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct ReferencedTokenStatus {
+    pub status_list: StatusListReference,
+}
+
+/// Claim describing the location of the Referenced Token in the Status List.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct StatusListReference {
+    pub idx: StatusListIndex,
+    pub uri: StatusListUri,
+}
+
+/// Index to check for status information in the Status List for the Referenced Token.
+#[derive(Copy, Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct StatusListIndex(pub u32);
+
+/// The URI where one can fetch the Status List or Status List Token
+/// containing the status information about the ReferencedToken.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct StatusListUri(pub String);
+
+impl ReferencedTokenStatus {
+    pub fn new(idx: StatusListIndex, uri: StatusListUri) -> Self {
+        Self {
+            status_list: StatusListReference { idx, uri },
+        }
+    }
+}
+
+impl Claim for ReferencedTokenStatus {
+    fn key() -> Key {
+        Key::Integer(65535)
+    }
+}
+impl From<ReferencedTokenStatus> for Value {
+    fn from(value: ReferencedTokenStatus) -> Self {
+        Value::Map(value.into())
+    }
+}
+impl From<ReferencedTokenStatus> for BTreeMap<Value, Value> {
+    fn from(value: ReferencedTokenStatus) -> Self {
+        BTreeMap::from_iter([(StatusListReference::key().into(), value.status_list.into())])
+    }
+}
+impl TryFrom<Value> for ReferencedTokenStatus {
+    type Error = ClaimError;
+
+    fn try_from(value: Value) -> Result<Self, Self::Error> {
+        Ok(serde_cbor::value::from_value(value)?)
+    }
+}
+
+impl Claim for StatusListReference {
```

Comment:
Why do you need to implement `Claim` for `StatusListReference`? I know they use the term "claim" in the spec to talk about `idx` and `uri`, but I think the only real CWT claim is `status_list` (65535), is it not?

Reply:
Oh maybe I misinterpreted that 🤔 I read it thinking they are all supposed to be separate claims but maybe that's not the case. @cobward do you know/have thoughts?

Reply:
+1 to what @timothee-haudebourg said (except there's a slight error in the name). `status` (65535) is the only claim. Extracting a status should yield the following structure (CBOR-diagnostic notation):
```
      {
          "status_list": {
            "idx": "0",
            "uri": "https://example.com/statuslists/1"
          }
      }
 ```
 
 So I would suggest having a structs like this:
 ```rust
 pub struct Status {
      status_list: StatusList,
 }
 
 pub struct StatusList {
      idx: u32,
      uri: String,
 }
 
 impl Claim for Status { ... }
 ```

Reply:
Possible even something like this to handle spec extensions gracefully:

```rust
pub struct Status {
     pub status_list: StatusList,
     pub extra: BTreeMap<String, serde_cbor::Value>,
}
```

Reply:
ahh I missed the spec extension comment earlier -- left it out for now but can add it in if needed!